component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -134,16 +138,73 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
         for (UserOperationEventListener listener : TokenExchangeServiceComponent.getUserOperationEventListeners()) {
             try {
                 listener.doPostAuthenticate(tokReqMsgCtx.getAuthorizedUser().getUserName(), false,
                         userStoreManager);
             } catch (UserStoreException e) {
+                if (e.getCause() instanceof AccountLockException) {
+                    String errorMessage = "Local user authorization failed: linked local account with id " + userId +
+                            " is locked";
+
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder =
+                                new DiagnosticLog.DiagnosticLogBuilder(
+                                        Constants.LogConstants.COMPONENT_ID,
+                                        Constants.LogConstants.ActionIDs.AUTHORIZE_LINKED_LOCAL_USER
+                                );
+                        diagnosticLogBuilder
+                                .resultMessage(errorMessage)
+                                .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                                .resultStatus(DiagnosticLog.ResultStatus.FAILED);
+                        LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+                    }
+
+                    handleException(OAuth2ErrorCodes.ACCESS_DENIED, errorMessage);
+                }
+
                 if (e.getCause() instanceof IdentityEventException) {
-                    handleException(OAuth2ErrorCodes.ACCESS_DENIED, "Local user authorization failed: " +
-                            e.getCause().getLocalizedMessage());
+                    String errorMessage = "Local user authorization failed for user: " + userId +
+                            " cause: " + e.getCause().getLocalizedMessage();
+
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder =
+                                new DiagnosticLog.DiagnosticLogBuilder(
+                                        Constants.LogConstants.COMPONENT_ID,
+                                        Constants.LogConstants.ActionIDs.AUTHORIZE_LINKED_LOCAL_USER
+                                );
+                        diagnosticLogBuilder
+                                .resultMessage(errorMessage)
+                                .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                                .resultStatus(DiagnosticLog.ResultStatus.FAILED);
+                        LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+                    }
+
+                    handleException(OAuth2ErrorCodes.ACCESS_DENIED, errorMessage);
                 }
+
+                if (LoggerUtils.isDiagnosticLogsEnabled()) {
```

Comment:
Add the log only for userStoreClientException

Reply:
Addressed with [65ee88f](https://github.com/wso2-extensions/identity-oauth2-grant-token-exchange/pull/21/commits/65ee88f94d62dd0869e4a9d8f475cd453a2c8278)